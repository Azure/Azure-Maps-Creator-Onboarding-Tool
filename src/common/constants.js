export const PATHS = {
  INDEX: '/',
  CREATE_MANIFEST: '/create-manifest',
  EDIT_MANIFEST: '/edit-manifest',
  PROCESSING: '/processing',
  CREATE_GEOREFERENCE: '/create-georeference',
  LAYERS: '/layers',
  LEVELS: '/levels',
  INVALID_PATH: '/*',
};

export const ROUTE_NAME_BY_PATH = {
  [PATHS.INDEX]: 'home',
  [PATHS.CREATE_MANIFEST]: 'create',
  [PATHS.EDIT_MANIFEST]: 'edit',
  [PATHS.PROCESSING]: 'processing',
  [PATHS.CREATE_GEOREFERENCE]: 'create',
  [PATHS.LAYERS]: 'create',
  [PATHS.LEVELS]: 'create',
  [PATHS.INVALID_PATH]: 'redirect',
};

export const CONSTANTS = {
  GEO: {
    US: {
      TEXT: 'geography.unitedstates',
      URL: 'https://us.atlas.microsoft.com',
    },
    // remove before moving to github
    US_TEST: {
      TEXT: 'geography.unitedstates.test',
      URL: 'https://us.t-azmaps.azurelbs.com',
    },
    EU: {
      TEXT: 'geography.europe',
      URL: 'https://eu.atlas.microsoft.com',
    }
  }
};

export const TRUNCATE_FRACTION_DIGITS = 8;