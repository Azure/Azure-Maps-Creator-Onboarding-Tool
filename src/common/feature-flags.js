const flags = {
  onboardingEnabled: false,
};

export default flags;