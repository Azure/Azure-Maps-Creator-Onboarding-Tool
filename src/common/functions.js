/**
 * Given a single path name, returns a list of all path names from root to the given path name.
 * Ex. Given '/create/hello', returns ['/', '/create', '/create/hello']
 * @param {String} pathName the path name of the URL
 * @returns {String[]} an array of path names
 */
export const getSplitPaths = (pathName) => {
  if (typeof pathName !== 'string' || pathName === '') { return []; }
  if (pathName === '/') { return ['/']; }

  const pathNameLowerCase = pathName.toLowerCase();
  const splits = pathNameLowerCase.split('/');
  const splitPaths = [];
  for (let i = splits.length - 1; i > 0; i--) {
    let stringBuilder = '';
    for (let j = 1; j <= i; j++) {
      stringBuilder += '/' + splits[j];
    }
    splitPaths.push(stringBuilder);
  }
  splitPaths.push('/');
  splitPaths.reverse();

  return splitPaths;
};