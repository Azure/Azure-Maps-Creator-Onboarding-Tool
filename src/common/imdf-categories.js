export const imdfCategories = [
  'auditorium',
  'brick',
  'classroom',
  'column',
  'concrete',
  'conferenceroom',
  'drywall',
  'elevator',
  'escalator',
  'fieldofplay',
  'firstaid',
  'fitnessroom',
  'foodservice',
  'footbridge',
  'glass',
  'huddleroom',
  'kitchen',
  'laboratory',
  'library',
  'lobby',
  'lounge',
  'mailroom',
  'mothersroom',
  'movietheater',
  'movingwalkway',
  'nonpublic',
  'office',
  'opentobelow',
  'parking',
  'phoneroom',
  'platform',
  'privatelounge',
  'ramp',
  'recreation',
  'restroom',
  'restroom.family',
  'restroom.female',
  'restroom.female.wheelchair',
  'restroom.male',
  'restroom.male.wheelchair',
  'restroom.transgender',
  'restroom.transgender.wheelchair',
  'restroom.unisex',
  'restroom.unisex.wheelchair',
  'restroom.wheelchair',
  'road',
  'room',
  'serverroom',
  'shower',
  'smokingarea',
  'stairs',
  'steps',
  'storage',
  'structure',
  'terrace',
  'theater',
  'unenclosedarea',
  'unspecified',
  'vegetation',
  'waitingroom',
  'walkway',
  'walkway.island',
  'wood',
];
