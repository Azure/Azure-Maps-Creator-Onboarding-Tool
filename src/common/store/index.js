export { getDomain, useUserStore } from './user.store';
export { useResponseStore } from './response.store';
export { progressBarSteps, progressBarStepsByKey, useCompletedSteps, useProgressBarStore } from './progress-bar-steps';
export { useDissolvedExterior, useGeometryStore } from './geometry.store';
export { useLayersStore } from './layers.store';
export { useLevelsStore } from './levels.store';