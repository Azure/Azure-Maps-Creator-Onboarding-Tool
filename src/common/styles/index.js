import { color, fontSize, fontWeight, strings } from './styles';

export { color, fontSize, fontWeight, strings };