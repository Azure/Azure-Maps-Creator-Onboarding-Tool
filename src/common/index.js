import { CONSTANTS, PATHS, ROUTE_NAME_BY_PATH } from './constants';

export { CONSTANTS, PATHS, ROUTE_NAME_BY_PATH };