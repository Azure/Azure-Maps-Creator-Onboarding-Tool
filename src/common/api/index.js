// The following API is designed to be used only by the onboarding tool and is not supported for any other use.
import { getEnvs } from 'common/functions';
import { useUserStore } from '../store/user.store';

export const uploadFile = (file) => {
  const { geography, subscriptionKey } = useUserStore.getState();
  const url = `${getEnvs()[geography].URL}/manifest?api-version=2.0&subscription-key=${subscriptionKey}`;

  return fetch(url, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/zip',
    },
    body: file,
  });
};

export const fetchFromLocation = (location) => {
  const { subscriptionKey } = useUserStore.getState();
  return fetch(`${location}&subscription-key=${subscriptionKey}`, {
    method: 'GET',
  });
};

export const deleteFromLocation = (fetchUrl) => {
  const { subscriptionKey } = useUserStore.getState();
  fetch(`${fetchUrl}&subscription-key=${subscriptionKey}`, {
    method: 'DELETE',
  });
};

export const fetchAddress = (address) => {
  const { geography, subscriptionKey } = useUserStore.getState();
  return fetch(`${getEnvs()[geography].URL}/search/address/json?subscription-key=${subscriptionKey}&api-version=1.0&query=${address}&limit=1`)
    .then((res) => res.json());
};