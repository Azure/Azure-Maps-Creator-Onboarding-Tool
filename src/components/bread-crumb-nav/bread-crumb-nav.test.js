import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import BreadCrumbNav from './bread-crumb-nav';
import { PATHS } from 'common';

describe('BreadCrumbNav', () => {
  it('should render', () => {
    render(
      <MemoryRouter initialEntries={[PATHS.INDEX]}>
        <BreadCrumbNav />
      </MemoryRouter>
    );

    // Home
    let homeButton = screen.getAllByText('home')[0];
    let createButtons = screen.queryAllByText('create');
    expect(homeButton).toBeInTheDocument();
    expect(createButtons).toEqual([]);

    // Redirect test (itself)
    fireEvent.click(homeButton);

    homeButton = screen.getAllByText('home')[0];
    createButtons = screen.queryAllByText('create');
    expect(homeButton).toBeInTheDocument();
    expect(createButtons).toEqual([]);
  });

  it('should render - create', async () => {
    render(
      <MemoryRouter initialEntries={[PATHS.CREATE_MANIFEST]}>
        <BreadCrumbNav />
      </MemoryRouter>
    );

    // Home > Create
    let homeButton = screen.getAllByText('home')[0];
    let createButtons = screen.getAllByText('create')[0];
    expect(homeButton).toBeInTheDocument();
    expect(createButtons).toBeInTheDocument();

    // Redirect test (itself)
    fireEvent.click(createButtons);

    homeButton = screen.getAllByText('home')[0];
    createButtons = screen.getAllByText('create')[0];
    expect(homeButton).toBeInTheDocument();
    expect(createButtons).toBeInTheDocument();

    // Redirect test (Home)
    fireEvent.click(homeButton);

    homeButton = screen.getAllByText('home')[0];
    createButtons = screen.queryAllByText('create');
    expect(homeButton).toBeInTheDocument();
    expect(createButtons).toEqual([]);
  });

  it('should match snapshot - edit', () => {
    expect(render(
      <MemoryRouter initialEntries={[PATHS.EDIT_MANIFEST]}>
        <BreadCrumbNav />
      </MemoryRouter>)
    ).toMatchSnapshot();
  });
});