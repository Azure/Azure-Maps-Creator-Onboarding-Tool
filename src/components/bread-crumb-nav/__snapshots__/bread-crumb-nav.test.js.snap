// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumbNav should match snapshot - edit 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          style="position: relative;"
        >
          <div
            data-automation-id="visibleContent"
            style="position: fixed; visibility: hidden;"
          >
            <div
              aria-label="navigational.breadcrumb"
              class="ms-Breadcrumb root-108"
              role="navigation"
            >
              <div
                class="ms-FocusZone css-116"
                data-focuszone-id="FocusZone5"
              >
                <ol
                  class="ms-Breadcrumb-list list-109"
                >
                  <li
                    class="ms-Breadcrumb-listItem listItem-110"
                  >
                    <button
                      class="ms-Link ms-Breadcrumb-itemLink itemLink-117"
                      tabindex="0"
                      type="button"
                    >
                      <div
                        class="ms-TooltipHost root-118"
                        role="none"
                      >
                        home
                        <div
                          hidden=""
                          id="tooltip6"
                          style="position: absolute; width: 1px; height: 1px; margin: -1px; padding: 0px; border: 0px; overflow: hidden; white-space: nowrap;"
                        >
                          home
                        </div>
                      </div>
                    </button>
                    <i
                      aria-hidden="true"
                      class="ms-Breadcrumb-chevron chevron-121"
                      data-icon-name="ChevronRight"
                    >
                      
                    </i>
                  </li>
                  <li
                    class="ms-Breadcrumb-listItem listItem-110"
                  >
                    <button
                      class="ms-Link ms-Breadcrumb-itemLink itemLink-117"
                      tabindex="-1"
                      type="button"
                    >
                      <div
                        class="ms-TooltipHost root-118"
                        role="none"
                      >
                        edit
                        <div
                          hidden=""
                          id="tooltip7"
                          style="position: absolute; width: 1px; height: 1px; margin: -1px; padding: 0px; border: 0px; overflow: hidden; white-space: nowrap;"
                        >
                          edit
                        </div>
                      </div>
                    </button>
                  </li>
                </ol>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        style="position: relative;"
      >
        <div
          data-automation-id="visibleContent"
          style="position: fixed; visibility: hidden;"
        >
          <div
            aria-label="navigational.breadcrumb"
            class="ms-Breadcrumb root-108"
            role="navigation"
          >
            <div
              class="ms-FocusZone css-116"
              data-focuszone-id="FocusZone5"
            >
              <ol
                class="ms-Breadcrumb-list list-109"
              >
                <li
                  class="ms-Breadcrumb-listItem listItem-110"
                >
                  <button
                    class="ms-Link ms-Breadcrumb-itemLink itemLink-117"
                    tabindex="0"
                    type="button"
                  >
                    <div
                      class="ms-TooltipHost root-118"
                      role="none"
                    >
                      home
                      <div
                        hidden=""
                        id="tooltip6"
                        style="position: absolute; width: 1px; height: 1px; margin: -1px; padding: 0px; border: 0px; overflow: hidden; white-space: nowrap;"
                      >
                        home
                      </div>
                    </div>
                  </button>
                  <i
                    aria-hidden="true"
                    class="ms-Breadcrumb-chevron chevron-121"
                    data-icon-name="ChevronRight"
                  >
                    
                  </i>
                </li>
                <li
                  class="ms-Breadcrumb-listItem listItem-110"
                >
                  <button
                    class="ms-Link ms-Breadcrumb-itemLink itemLink-117"
                    tabindex="-1"
                    type="button"
                  >
                    <div
                      class="ms-TooltipHost root-118"
                      role="none"
                    >
                      edit
                      <div
                        hidden=""
                        id="tooltip7"
                        style="position: absolute; width: 1px; height: 1px; margin: -1px; padding: 0px; border: 0px; overflow: hidden; white-space: nowrap;"
                      >
                        edit
                      </div>
                    </div>
                  </button>
                </li>
              </ol>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
